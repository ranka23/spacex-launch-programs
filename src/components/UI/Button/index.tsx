import React from "react";
import styles from "./styles.module.scss";

interface ToggleButtonProps {
  toggleValue: string | boolean;
  onClick: (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => void | undefined;
  buttonKey?: string;
  className?: string;
}

const Button: React.FC<ToggleButtonProps> = ({
  toggleValue,
  onClick,
  buttonKey = "",
  className = "",
}: ToggleButtonProps) => {
  return (
    <button
      key={buttonKey}
      type="button"
      onClick={onClick}
      className={`${styles.button_container} ${className}`}
    >
      {toggleValue.toString()}
    </button>
  );
};

Button.defaultProps = {
  buttonKey: "",
  className: "",
};

export default Button;
