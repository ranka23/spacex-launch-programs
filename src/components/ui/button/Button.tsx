/* eslint-disable react/require-default-props */
import React from "react";
import styles from "./button.module.scss";

interface ToggleButtonProps {
  toggleValue: string | boolean;
  onClick?: (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => void | undefined;
  buttonKey?: string;
  className?: string;
}

const Button: React.FC<ToggleButtonProps> = ({
  toggleValue,
  onClick,
  buttonKey = "",
  className = "",
}: ToggleButtonProps) => {
  return (
    <button
      key={buttonKey}
      type="button"
      onClick={onClick}
      className={`${styles.button_container} ${className}`}
    >
      {toggleValue.toString()}
    </button>
  );
};
export default Button;
