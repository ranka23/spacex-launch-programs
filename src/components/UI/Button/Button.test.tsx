import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Button from "./index";

describe("<Button />", () => {
  const tree = () =>
    render(
      <MemoryRouter>
        <Button
          onClick={() => {
            console.log("Hello World");
          }}
          toggleValue="2020"
        />
      </MemoryRouter>
    ).container.firstChild;

  it("renders props with no errors", () => {
    expect(tree()).toMatchSnapshot();
  });
});
