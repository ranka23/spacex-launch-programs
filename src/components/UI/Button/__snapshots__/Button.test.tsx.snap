// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders props with no errors 1`] = `
<button
  class="button "
  type="button"
>
  2020
</button>
`;
